### ISSUE TITLE
Set status "archived" disappered from API

### ISSUE THREAD

**Issue by BenjV** [SiCKRAGETV/sickrage-issues#3329]
SR Version:  BRANCH: (master) / COMMIT: (136f531f93093a5287b588ca37e19c11153f47a4)  

I noticed that the set episode status call to the API, to change the status of an episode to "Arcived" does not work anymore.
Is this intentional or will this be corrected?
I used this in my own program to change the status to archived when I move the episode outside the controle of Sickrage.

Also were is the documentation for the API, now that the api builder is made "fancy" but much less informative.

**Comment by SiCKRAGEB0T** [comment]
## Thanks for the bug/issue report
Please make sure to submit all of the [required information](https://github.com/SiCKRAGETV/sickrage-issues/blob/master/README.md), or we won't be able to help you!
Provide as **much** information as you can such as debug logs, commit hash, and tell us how to replicate the problem.For basic **questions** use the [ [forum] ](https://sickrage.tv/forums/), and for **Feature Requests** [ [Feathub] ](http://feathub.com/SiCKRAGETV/SickRage)
Also make sure the title accurately describes your issue. ("SR not working" etc. is not ok!)

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist)

## Many issues can simply be solved by:
- Making sure you update to the latest version.
- Have you tried turning your device off and on again?
- Using the **search** function to see if this issue has already been reported/solved.
- Do a [GIT reset](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#update-problems-try-this).
- Checking the [ [Wiki] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki) for [ [Guides] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Installation-&-Configuration-Guides), [ [FAQS] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes) [ [Main settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Settings-explained), [ [Show settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Show-settings-explained) [ [Remaining settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Remaining-settings-explained)
- Shutting down SickRage and removing the `cache/mako` & `cache/sessions` folders.
- If SickRage doesnt find any episodes you probably need to :
 - add a [Scene Exception](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Scene-exceptions-and-numbering)
 - Set the [Default Episode Status](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#newly-aired-shows-are-not-downloading-and-set-to-skippedignored) to `WANTED`
 - Make sure you understand how [quality detection](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Quality-Settings#quality-detectiondetermination) works during searches.
 - Shutdown SickRage and remove the `cache.db` file.
- If you use a QNAP and have SSL errors try this [package](http://apps.qnap.community/11-community/17-qsickrage).
- If you have a `'ascii' codec can't encode character` error, than set your Locale to [UTF-8](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#i-have-problems-with-special-characters-%C3%A9-etc-what-can-i-do)

*ps. If you enjoy Sickrage and like the support provided, than please consider making a [donation](https://github.com/SiCKRAGETV/SickRage/wiki/Donations) to keep SickRage alive.*

**Comment by fernandog** [comment]
@MGaetan89 we now use archive+quality

**Comment by BenjV** [comment]
But the api does not support Archive anymore.

And why + quality? Sickrage already knows the quality of the downloaded episode.
What is the usefulness of that?

**Comment by OmgImAlexis** [comment]
@BenjV we don't set it to just archived anymore, we set the episodes for example to `Archived SDTV` that way we know what quality it was archived at.

**Comment by fernandog** [comment]
No. When status archive SR lose download quality. Thats why
When you delete/move file SR auto change to <default deleted ep status>. Can be archive

**Comment by BenjV** [comment]
Ok fine but can we get back the possibility to change the status to "archived + quality" in the API?
I really like this function to return in the API calls.
And perhaps some documentation or the old api builder back, because the new one is pretty useless.
One has to execute api calls to be able to see the parameters, thats not the way to do it.

By the way, Sickrage does not change the status to "Can be Archive" at all when I move files out of the Sickrage control. I keeps the status "downloaded + quality"

**Comment by OmgImAlexis** [comment]
@BenjV you shouldn't have to execute and API calls to see the parameters, @MGaetan89

**Comment by MGaetan89** [comment]
I can look at this next time I refactor the API.
But as @OmgImAlexis said, you can see the documentation and parameters without having to execute the command. Just click on the command name and you can see everything you need.

**Comment by fernandog** [comment]
@BenjV so enable 'Skip Remove Detection'

You should read wiki explaining all SR settings.

**Comment by BenjV** [comment]
Thanks for the info, but if I click on the command name it only shows the first part of the command, not the parameter and its format.
It only shows this:
/api/<api number>/?cmd=episode.setstatus
 
And no status update to archived possible.
just 'wanted', 'skipped', 'ignored', 'failed'

**Comment by MGaetan89** [comment]
I am not sure I understand what you mean. Can you post a screenshot?
On my side I have all the information about the command:

<img width="1188" alt="episode.setstatus" src="https://cloud.githubusercontent.com/assets/1009664/10558595/46b7f962-74d6-11e5-89c9-6ad2537e9f10.png">

**Comment by fernandog** [comment]
@MGaetan89 yes, and it's pretty good! thank you for that

**Comment by BenjV** [comment]
Before this API change I could send the following command to the API.

cmd=episode.setstatus&indexerid=257655&season=01&episode=02&status=archived&force=1

This would change the status of Season 01, Episode 02 of the show Arrow (indexerid=257655) to archived.
In the old api builder this information was clearly to be found.
In the new apibuilder only the first part of this command is to be found.

You made nice pull downs of them and they look pretty but don't give any information to a programmer how to use them and an Api is to be used by other programs.
So maybe it would be a good idee to add the pulldown selection to the URL that is displayed below the word "Playground" or add a test button to see the command.

At the moment I have to execute the api call to be able to see the whole command.

**Comment by MGaetan89** [comment]
So you want to see the full URL without running the command, right? I can add that, it's a good point.

I found two other things that should be fixed in the API builder:
- When selecting a season, "Season 0" should be "Specials"
- When the parameter `season` is required, but `episode` is optional, it is not possible to select an episode (like for `episode.setstatus`)

**Comment by OmgImAlexis** [comment]
@MGaetan89 another thing would be to change the way the dropdowns are shown, currently it looks like the right dropdown sets the value of the left dropdown. For example here it looks like the field `status` is being set to indexerid` and the field `force` is being set to `tvdbid`.
Also shouldn't `tvdbid` be `thetbdbid`?

![screen shot 2015-10-18 at 2 14 35 am](https://cloud.githubusercontent.com/assets/6525926/10559660/01bc618c-753e-11e5-8195-c90d717ddc1e.png)

**Comment by MGaetan89** [comment]
@OmgImAlexis I don't agree. This is how dropdowns works everywhere. They just display their own value, they don't set it to whatever is next to them.
And `tvdbid` is correct. That is how the parameter is named.

**Comment by OmgImAlexis** [comment]
No dropdown we have works like that....?

![screen shot 2015-10-18 at 11 39 31 am](https://cloud.githubusercontent.com/assets/6525926/10562008/f0738960-758c-11e5-9cf7-4cc3a369509b.png)
![screen shot 2015-10-18 at 11 39 19 am](https://cloud.githubusercontent.com/assets/6525926/10562009/f0a3c418-758c-11e5-8280-e3e2d64b7b84.png)
![screen shot 2015-10-18 at 11 39 09 am](https://cloud.githubusercontent.com/assets/6525926/10562010/f0a93c86-758c-11e5-86d6-fd93a3511b9f.png)

**Comment by MGaetan89** [comment]
That would be too much for a command with many parameters (see `show.addnew` for example). It's already messy enough now, without the labels.
I could put the dropdown inside the "Parameters" table. But again, for commands like `show.addnew` it would compact everything a lot I guess.

**Comment by BenjV** [comment]
Maybe just every parameter on a new line, so the suggestion that they belong together on the same line is gone?

**Comment by MGaetan89** [comment]
Again, for commands like `show.addnew` it would be really unusable.

**Comment by miigotu** [comment]
@MGaetan89 we could add back a method with `archived` as the parameter, and it would have to only be allowed on a file in downloaded status (aka you split the status and quality, and append the old quality to archived status and set it to that compound status)

Was thinking to adjust the episode status management and displayShow droppdowns to do this for archived and downloaded statuses so there is only one option to be made in the dropdown instead of one option for each quality.

**Comment by BenjV** [comment]
Would be great.

**Comment by fernandog** [comment]
user should have the option to manually change archive quality

useful when you have a show that you deleted from SR after show was cancelled but other network bought so you re-add again OR when you lost all your files and creating a new db.
 As you don't have the files, SR won't detect it and they will be unkonwn. So you manually mark as archived + quality
